### DISCUSSION TITLE
Idea for a possible new feature: search RPM files?

### DISCUSSION THREAD

**Discussion by genivia-inc** [Genivia/ugrep#365] [Ideas]
Would there be any interest to search RPM files?

RPM files contain [CPIO payloads](https://refspecs.linuxbase.org/LSB_3.1.1/LSB-Core-generic/LSB-Core-generic/pkgformat.html). Searching CPIO archives is already supported. To search RPM, we probably want to search the CPIO payloads only and ignore the RPM headers. In this way, (compressed) RPM will be searched similar to tarballs and (compressed) CPIO files.

If there is interest and practical use cases for this feature, then let's move forward to add this to ugrep.

**Comment by vt-alt** [comment]
You will still need to parse RPM headers to determine payload compression method.
